sorting/tim_sort.cpp

```diff
@@ -12,7 +12,7 @@ void insertionSort(int arr[], int left, int right) {
     for (int i = left + 1; i <= right; i++) {
         const int temp = arr[i];
         int j = i - 1;
-        while (arr[j] > temp && j >= left) {
+        while (j >= left && arr[j] > temp) {
```

Comment:
What is the difference?

Reply:
[Short-circuit_evaluation](https://en.wikipedia.org/wiki/Short-circuit_evaluation).

The problem was that `j` could become `-1`. In the old code there was was a check `arr[-1] > temp`, which lead to _stack buffer underflow_. In the new code, you first check if `j >= left`, and it this is true, you check if `arr[j] > temp`.